cmd/debugmailbox/src/lib.rs

```diff
@@ -64,36 +68,82 @@ enum DebugMailboxCmd {
     Debug,
     /// Force the device into ISP mode
     Isp,
+    /// Get a Debug Authentication Challenge
+    AuthChallenge { out: PathBuf },
+    /// Send a Debug Authentication Response
+    AuthResponse { dar: PathBuf },
 }
 
-fn alive<'a>(
+fn poll_raw_ap_register<'a, F>(
```

Comment:
FWIW, you could eliminate the parameters and write this as

```rust
fn poll_raw_ap_register(
    probe: &mut Box<dyn ArmProbeInterface + '_>,
    ap: &ApAddress,
    addr: u8,
    f: impl Fn(u32) -> Result<bool>,
    timeout_ms: usize,
) -> Result<u32> {
```
I think this is preferable because
- it doesn't introduce the `'a` named lifetime just to use it once (I went searching to see where the other use was)
- I personally find `impl Trait` on parameters easier to read than separate where clauses; in cases where the parameter (here, `F`) appears only once they're basically equivalent to the compiler

For generality I'd also consider changing `Fn` to `FnMut`, which would allow the caller to e.g. maintain a counter. (That will also require you to `mut f`.)

Reply:
oh, that is nice.  I tried to do this initially but kept getting hung up on the probe lifetime.  Using an anonymous lifetime makes perfect sense.